ooniprobe/AppDelegate.mm

```diff
@@ -27,6 +27,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
     [[UIApplication sharedApplication] setStatusBarStyle:UIStatusBarStyleLightContent];
     
+    
+    //TODO register as a probe if I have never done it. This is needed to register the probe even if push notifications are turned off.
```

Comment:
I still don't know how to handle this. I need a way to understand if the user registered at least once to the notification.

Reply:
I think what we really need is some semantic like get-if-modified-since in the backend. Except that here it would be something like register-if-not-already-registered. Otherwise this is going to create many headaches for us.

Makes sense?

Reply:
To be more clear: it is my understanding that the current API is not idempotent in that you cannot ask to register a probe if it's not already registered and consider that an update otherwise. I'd really like to have a method with this semantic so that we can just issue such an idempotent request on load. This is particularly useful to us because there are cases in which some devices are used both for testing and for production. And, in general, having an idempotent register method is gonna be useful to us. I wonder whether like this it's more clear, @lorenzoPrimi what do you think?